### ISSUE TITLE
Thumbnail for small labels

### ISSUE THREAD

**Issue by adityaapte** [napari/napari#6707]
## 🚀 Feature
Crop labels to exclude background while showing thumbnails in layer list.

## Motivation
For radiological images, segmentation of tumor occupies a lot fewer voxels compared to the underlying CT scan. The thumbnails for such labels are then displayed as mostly black (background). 

## Pitch

Crop the mask for labels layer to the foreground extents before creating its thumbnail. 

## Alternatives

(1) Provide functionality for users to customize thumbnail creation.
(2) Provide functionality to set foreground and background colors for layer list entries.

## Additional context
<img width="952" alt="image" src="https://github.com/napari/napari/assets/840839/f0cc6bc5-80fe-472e-8cff-11c2ab2b21c7">

**Comment by psobolewskiPhD** [comment]
Thanks for opening the issue.
I agree that our thumbnails can be less than useful in many cases due to how small they are and how much downsampling occurs. They are supposed to give a birds-eye view of the layer, but yeah...

Here's another example, where I annotate the Astronaut sample image.
The Shapes thumbnail is ???, Points is useless because of downsampling, and the Labels one is barely useful?
<img width="1134" alt="image" src="https://github.com/napari/napari/assets/76622105/ebc91a36-ac4e-4e3a-ab29-70d3c20c0ab1">

There is some discussion here:
https://napari.zulipchat.com/#narrow/stream/212875-general/topic/Thumbnails/near/261301624

Basically, at present we lack a good idea of what to change and overall thumbnails are lower priority vs. the main canvas.
Perhaps this is an area where a plugin could provide an improved thumbnail widget that would be larger basically?
A related, nice-to-have would be a minimap where you could see what area you are viewing of the larger layer extent, ideally being able to click it to jump to the location.
